sdk/core/azure-core/src/http/policy_retry.cpp

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <http/policy.hpp>
+
+#include <algorithm>
+#include <cstdlib>
+#include <limits>
+#include <thread>
+
+using namespace Azure::Core::Http;
+
+namespace {
+decltype(RetryOptions::RetryDelay) ShouldRetry(
+    Response const& response,
+    RetryOptions const& retryOptions,
+    decltype(RetryOptions::MaxRetries) attempt)
+{
+  // Are we out of retry attempts?
+  if (attempt > retryOptions.MaxRetries)
+  {
+    return std::chrono::seconds(-1);
+  }
+
+  // Should we retry on the given response retry code?
+  auto const& statusCodes = retryOptions.StatusCodes;
+  auto const statusCodesEnd = statusCodes.end();
+  if (std::find(statusCodes.begin(), statusCodesEnd, response.GetStatusCode()) == statusCodesEnd)
+  {
+    return std::chrono::seconds(-1);
+  }
+
+  // Try to find retry-after headers. There are several of them possible.
+  decltype(response.GetHeaders()) responseHeaders = response.GetHeaders();
+  auto const responseHeadersEnd = responseHeaders.end();
+  auto header = responseHeadersEnd;
+  if (((header = responseHeaders.find("retry-after-ms")) != responseHeadersEnd)
+      || ((header = responseHeaders.find("x-ms-retry-after-ms")) != responseHeadersEnd))
+  {
+    // The headers above are in milliseconds.
+    return std::chrono::milliseconds(std::stoi(header->second));
+  }
+
+  if ((header = responseHeaders.find("Retry-After")) != responseHeadersEnd)
+  {
```

Comment:
It would be best to separate the retry policies here. This is a different retry policy than `exponential_retry_policy`.

Reply:
I'll see what I can do, if you're talking about separating to different functions (but to the same effect - because this is what other SDKs end up doing).

Reply:
I split that single functions into a few smaller ones of which each one does its only job.